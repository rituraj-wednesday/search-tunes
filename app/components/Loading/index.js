/**
 *
 * Loading
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

/**
 * Loading component displays loading animation.
 *
 * @date 09/05/2024 - 16:37:00
 *
 * @param {Object} props - The component props.
 * @param {string} props.height - height prop.
 * @param {string} props.width - width prop.
 * @param {string} props.fill - fill prop.
 * @returns {JSX.Element} The Loading component.
 */
function Loading({ height, width, fill }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      x="0px"
      y="0px"
      width={width}
      height={height}
      viewBox="0 0 50 50"
      xmlSpace="preserve"
      aria-label="loading"
    >
      <path
        fill={fill}
        d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
      >
        <animateTransform
          attributeType="xml"
          attributeName="transform"
          type="rotate"
          from="0 25 25"
          to="360 25 25"
          dur="0.6s"
          repeatCount="indefinite"
        />
      </path>
    </svg>
  );
}

Loading.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
  fill: PropTypes.string
};

Loading.defaultProps = {
  height: '40px',
  width: '40px',
  fill: '#000'
};

export default Loading;
