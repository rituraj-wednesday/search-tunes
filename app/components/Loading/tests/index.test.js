/**
 *
 * Tests for Loading
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Loading from '../index';

describe('<Loading />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Loading />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Loading component', () => {
    const { getAllByLabelText } = renderWithIntl(<Loading />);
    expect(getAllByLabelText('loading').length).toBe(1);
  });
});
