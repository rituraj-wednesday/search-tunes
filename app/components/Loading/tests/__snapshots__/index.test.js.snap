// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render and match the snapshot 1`] = `
<body>
  <div>
    <svg
      aria-label="loading"
      height="40px"
      viewBox="0 0 50 50"
      width="40px"
      x="0px"
      xml:space="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
        fill="#000"
      >
        <animatetransform
          attributeName="transform"
          attributeType="xml"
          dur="0.6s"
          from="0 25 25"
          repeatCount="indefinite"
          to="360 25 25"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</body>
`;
